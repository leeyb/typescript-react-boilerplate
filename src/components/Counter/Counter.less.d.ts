export const container: string;
export const value: string;
export const btn: string;
