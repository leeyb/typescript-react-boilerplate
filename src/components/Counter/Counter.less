.container
{
    background-color: #ddd;

    .value
    {
        display: inline-block;
        min-width: 30px;
        text-align: center;
    }

    .btn
    {
        margin-left: 15px;
    }
}
