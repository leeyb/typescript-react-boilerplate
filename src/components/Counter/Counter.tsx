import * as cs from "classnames";
import * as React from "react";
import { SFC } from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";

import { ReduxProps } from "../../redux/actionHelper";
import { actionCreators, CounterActionCreators, CounterState } from "../../redux/counter";
import { RootState } from "../../redux/rootReducer";

import * as styles from "./Counter.less";

/**
 * Represents the props of the `Counter` component.
 */
export interface CounterProps extends ReduxProps<CounterActionCreators>, CounterState
{
}

/**
 * The `Presentational` `Counter` component.
 */
export const Counter: SFC<CounterProps> = (props) =>
{
    const { className, actions: { increase, decrease, increaseAsync, decreaseAsync }, value } = props;
    return (
        <div className={cs(className, styles.container)}>
            <span className={styles.value}>{value}</span>
            <button className={styles.btn} onClick={() => increase(1)}>+1</button>
            <button className={styles.btn} onClick={() => decrease(1)}>-1</button>
            <button className={styles.btn} onClick={() => increase(2)}>+2</button>
            <button className={styles.btn} onClick={() => decrease(3)}>-3</button>
            <button className={styles.btn} onClick={() => increaseAsync(1)}>Asynchronous Increment</button>
            <button className={styles.btn} onClick={() => decreaseAsync(1)}>Asynchronous Decrement</button>
        </div>
    );
};

/**
 * The `Container` `Counter` component.
 */
export default connect(
    (state: RootState) =>
    {
        return { ...state.counter };
    },
    (dispatch: Dispatch<CounterState>) =>
    {
        return { actions: bindActionCreators(actionCreators, dispatch) };
    }
)(Counter);
