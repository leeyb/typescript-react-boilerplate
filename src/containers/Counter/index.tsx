// @ts-ignore
import * as React from "react";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";

import { actionCreators, CounterState } from "../../redux/counter";
import { RootState } from "../../redux/rootReducer";

// @ts-ignore
import Counter, { CounterProps } from "../../components/Counter";

/**
 * The `Container` `Counter` component.
 */
export default connect(
    (state: RootState) =>
    {
        return { ...state.counter };
    },
    (dispatch: Dispatch<CounterState>) =>
    {
        return { actions: bindActionCreators(actionCreators, dispatch) };
    }
)(Counter);
