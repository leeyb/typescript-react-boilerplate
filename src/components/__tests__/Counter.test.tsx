import { shallow, ShallowWrapper } from "enzyme";
import * as React from "react";

import { Counter, CounterProps } from "../Counter";

describe("Counter (Presentational)", () =>
{
    let props: CounterProps;
    let wrapper: ShallowWrapper<CounterProps>;

    beforeAll(() =>
    {
        props = {
            actions: {
                decrease: jest.fn(),
                decreaseAsync: jest.fn(),
                increase: jest.fn(),
                increaseAsync: jest.fn()
            },
            className: "counter-test",
            value: 0
        };
        wrapper = shallow(<Counter {...props} />);
    });

    it("render", () =>
    {
        expect(wrapper).toMatchSnapshot();
    });

    it("init value", () =>
    {
        expect(wrapper.find(".value").text()).toEqual(String(props.value));
    });

    it("class name", () =>
    {
        expect(wrapper.find(`.${props.className}`)).not.toBeNull();
    });
});
