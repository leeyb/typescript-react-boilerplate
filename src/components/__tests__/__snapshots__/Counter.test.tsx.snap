// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter (Presentational) render 1`] = `
<div
  className="counter-test container"
>
  <span
    className="value"
  >
    0
  </span>
  <button
    className="btn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    -1
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    +2
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    -3
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    Asynchronous Increment
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    Asynchronous Decrement
  </button>
</div>
`;
